ocaml/utils/language_extension.ml

```diff
@@ -111,77 +139,98 @@ let equal a b = Option.is_some (equal_t a b)
 (* extension universes *)
 
 module Universe : sig
-  val is_allowed : 'a t -> bool
-
-  val check : 'a t -> unit
+  val is_allowed : extn_pair -> bool
 
-  val check_maximal : unit -> unit
+  val check : extn_pair -> unit
 
   type t =
     | No_extensions
-    | Only_erasable
-    | Any
+    | Upstream_compatible
+    | Stable
+    | Beta
+    | Alpha
+
+  val all : t list
+
+  val to_string : t -> string
 
-  val set : t -> bool
+  val of_string : string -> t option
+
+  val set : t -> unit
 
   val is : t -> bool
 end = struct
   (** Which extensions can be enabled? *)
   type t =
     | No_extensions
-    | Only_erasable
-    | Any
+    | Upstream_compatible
+    | Stable
+    | Beta
+    | Alpha
+  (* If you add a constructor, you should also add it to [all]. *)
+
+  let all = [No_extensions; Upstream_compatible; Stable; Beta; Alpha]
+
+  let to_string = function
+    | No_extensions -> "no_extensions"
+    | Upstream_compatible -> "upstream_compatible"
+    | Stable -> "stable"
+    | Beta -> "beta"
+    | Alpha -> "alpha"
+
+  let of_string = function
+    | "no_extensions" -> Some No_extensions
+    | "upstream_compatible" -> Some Upstream_compatible
+    | "stable" -> Some Stable
+    | "beta" -> Some Beta
+    | "alpha" -> Some Alpha
+    | _ -> None
 
   let compare t1 t2 =
-    let rank = function No_extensions -> 1 | Only_erasable -> 2 | Any -> 3 in
+    let rank = function
+      | No_extensions -> 0
+      | Upstream_compatible -> 1
+      | Stable -> 2
+      | Beta -> 3
+      | Alpha -> 4
+    in
     compare (rank t1) (rank t2)
 
-  let universe = ref Any
+  (* For now, the default universe is set to [Alpha] but only a limited set
+     of extensions is enabled. After the migration to extension universes,
+     the default will be [No_extensions]. *)
```

Comment:
I seem to remember something going by saying the default would be `Stable`. Not very important for this PR, but we should make sure to sync on this. Check our design doc; I may be misremembering.

Reply:
The default set in build rules would be `Stable`.